### PR TITLE
feat(ci-builder): Use abigenrc

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Updates the ci-builder to use the `.abigenrc` file and
then installs `abigen` using this version tag.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6556** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6556" target="_blank">Graphite</a> 
    * **PR #6558** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6558" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6558?utm_source=stack-comment).

**Review by clabby — APPROVED** [review]
LGTM, we build ci-builder with the docker context as the monorepo root, so should be fine.

File: ops/docker/ci-builder/Dockerfile [hunk 1]
```diff
@@ -11,10 +11,11 @@ RUN apt-get update && \
   chmod +x ./rustup.sh && \
   ./rustup.sh -y
 
-COPY ./.foundryrc ./.foundryrc
+COPY ./.abigenrc ./.abigenrc
```
Comment (trianglesphere) [inline ops/docker/ci-builder/Dockerfile:14]: Is this file checked in?

Comment (refcell) [reply ops/docker/ci-builder/Dockerfile:14]: Yes, please see the pr this is stacked ontop of #6556

**Review by tynes — APPROVED** [review]

**Pull request merged**